src/libsyntax/ext/tt/quoted.rs

```diff
@@ -84,12 +85,20 @@ pub enum KleeneOp {
 /// are "first-class" token trees. Useful for parsing macros.
 #[derive(Debug, Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash)]
 pub enum TokenTree {
-    Token(Span, token::Token),
+    Token(
+        Span,
+        token::Token,
+        bool, /* escape hygiene */
```

Comment:
It's probably better to make this an enum, `enum IdentHygiene { DefSite, CallSite }` or something.

Reply:
Not sure about that... seems like unnecessary complication, no?

Reply:
The hygiene escape affects only identifiers, but the field is present for all tokens, so something like `Token(span, some_separator_token, false)` in code above doesn't look especially good.
Is it very intrusive to place this flag into `Token::Ident` instead? (I don't know, it certainly may be.)

Reply:
>Not sure about that... seems like unnecessary complication, no?

This will certainly make life easier for any future reader.
Look at other stuff in data structures like AST, it mostly uses enums with meaningful names instead of booleans.

Reply:
>The hygiene escape affects only identifiers, but the field is present for all tokens

Nevermind, `quoted::TokenTree` is used only macro parsing and we want to keep `#` a detail of macro parsing as well, but `Token::Ident` hs much wider use, so the current implementation is okay.